### DISCUSSION TITLE
Alien light component have to reset to prefab for use

### DISCUSSION THREAD

**Discussion by jcx1991911** [FakeFishGames/Barotrauma#13806] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Any alien light component that already place in old sub no longer glow until you click the reset to prefab button.

### Reproduction steps

1.Find a submarine that was created before v1.3.0.4 that already placed the Alien Lighting Component
2.Enter the test or toggle visibility of lighting

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.4.3.0 (unstable)

### -

v1.4.4.1

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I'm afraid I could not reproduce this with any of the vanilla content. There are alien light components in some alien ruin modules (which are technically submarines), and they seem to be working correctly in all of those, so I suspect this has to do with how they've been set up in the specific sub you're testing. Could you give us an example sub (either by sending it as a .zip archive here or by linking to one in the Workshop) so we could give it a test? Thanks and sorry for the inconvenience!

**Reply by jcx1991911** [reply]
https://steamcommunity.com/sharedfiles/filedetails/?id=3014307661

It have two alien light component, one hidden in the center of the alien generator and the other under the workbench lamp, I'm sorry it's a shuttle, you'll need to use test ground in the outpost category for test (Included in the workshop file)

**Reply by Regalis11** [reply]
The issue seems to be that the component is set to be hidden in-game. It seems this is related to a fix we did in the latest update: there was a bug that caused the lights to show despite the item being hidden, which was not intentional, but I assume you were relying on this behavior to hide the sprite while keeping the emitted light visible. It should be possible to fix the problem by unchecking "hidden in game" and instead setting the sprite color to fully transparent (255,255,255,0). Sorry for the inconvenience!

**Reply by jcx1991911** [reply]
Okay, I didn't think that would be the reason, thanks for your patience in explaining it.
This change is important for decorating with lights.
